kotlin-android/app/src/main/java/com/mongodb/tasktracker/MemberActivity.kt

```diff
@@ -0,0 +1,116 @@
+package com.mongodb.tasktracker
+
+import android.app.AlertDialog
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.EditText
+import android.widget.Toast
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.mongodb.tasktracker.model.*
+import io.realm.Realm
+import io.realm.mongodb.functions.Functions
+import org.bson.Document
+import java.util.*
+
+class MemberActivity : AppCompatActivity() {
+    private var user: io.realm.mongodb.User? = null
+    private lateinit var realm: Realm
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var adapter: MemberAdapter
+    private lateinit var fab: FloatingActionButton
+    private lateinit var members: ArrayList<Member>
+
+    override fun onStart() {
+        super.onStart()
+        try {
+            user = taskApp.currentUser()
+        } catch (e: IllegalStateException) {
+            Log.w(TAG(), e)
+        }
+        if (user == null) {
+            // if no user is currently logged in, start the login activity so the user can authenticate
+            startActivity(Intent(this, LoginActivity::class.java))
+        } else {
+            // display the name of the project whose membership we're viewing in the action bar
+            val projectName = intent.extras?.getString(PROJECT_NAME_EXTRA_KEY)
+            title = projectName
+            setUpRecyclerView()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_member)
+
+        recyclerView = findViewById(R.id.project_users_list)
+        fab = findViewById(R.id.floating_action_button)
+        // create a dialog to add a user by email when an item is clicked
+        fab.setOnClickListener {
+            val input = EditText(this)
+            val dialogBuilder = AlertDialog.Builder(this)
+            dialogBuilder.setMessage("Add user to project by email:")
+                .setCancelable(true)
+                .setPositiveButton("Add User") { dialog, _ ->
+                    run {
```

Comment:
If you think it would be clearer, I can always put it in a separate function (lambda or otherwise). I think a lambda would fit best since the logic is very specific to this particular button, so it's not exactly reusable.

Reply:
There's a benefit to reducing spaghetti as well as reusability.

Reply:
Agreed. I trimmed this down a bit (removed the unnecessary ``run`` blocks), would you take another look? I'm on the fence if it would be harder or easier to follow the trimmed-down version if I pulled out the positive button handling into a lambda.